lib/swaps/Swaps.ts

```diff
@@ -435,15 +440,17 @@ class Swaps extends EventEmitter {
     if (quantity) {
       deal.quantity = quantity; // set the accepted quantity for the deal
       if (quantity <= 0) {
-        // TODO: accepted quantity must be a positive number, abort deal and penalize peer
+        this.setDealState(deal, SwapState.Error, 'accepted quantity must be a positive number');
+        await this.nodes.addReputationEvent(peer.nodePubKey!, ReputationEvent.SwapFailure);
+        return;
       } else if (quantity > deal.proposedQuantity) {
-        // TODO: accepted quantity should not be greater than proposed quantity, abort deal and penalize peer
+        this.setDealState(deal, SwapState.Error, 'accepted quantity should not be greater than proposed quantity');
+        await this.nodes.addReputationEvent(peer.nodePubKey!, ReputationEvent.SwapFailure);
```

Comment:
I don't think these reputation events need to be part of this PR, but if we do add them it should be in OrderBook.ts when we handle the `swap.failed` event (that way we don't need to have separate list of nodes in Swaps.ts). A separate PR would be better though just for a clearer commit history, we can penalize based on the reason for the failure. That might benefit from having #671 implemented first.

Reply:
Agree this should be in a different pr.